### PR TITLE
feat(kda): preserve FP32 beta logits in fused sigmoid

### PR DESCRIPTION
**Pull request by YangXu1990uiuc**
## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [x] I added GitHub labels: one `cat-*`, one or more `mod-*`, and one `orig-*` (see [label list](https://github.com/NVIDIA/cudnn-frontend/labels)).
- [ ] I set the **Milestone** and **Projects** fields in the sidebar (milestone is set; a maintainer must add the GitHub Project).

## Affected area

Python API or bindings

### Downstream model impact (combined attribution)

On B200, an official-count Kimi-K3 BF16 fprop+bprop feature-substack integration carrying patch-identical versions of both commits in this PR measured **1.3567x / 1.3238x / 1.2189x** at 8K/16K/32K, removing **26.3% / 24.5% / 18.0%** of the declared substack latency. The all-on arm combines merged #819 plus this raw-FP32-beta contract with three native causal-convolution calls and native SiTU; MLA is unchanged. This is not a marginal #802 or full-model result.

- Rebase the FP32-beta contract onto `develop` after #819, without carrying a duplicate KDA kernel implementation.
- Accept FP32 or I/O-dtype raw beta logits when KDA applies sigmoid inside the kernel, and return `dBeta` in the matching dtype.
- Preserve official Kimi's FP32 beta logits in the FLA adapter instead of rounding them to BF16.
- Cover both raw-beta dtypes in forward and backward tests, including the official fused Kimi call shape.

## Why

Official Kimi keeps raw beta logits in FP32 until the fused sigmoid. #819 generalized the FROST kernels to carry beta's element type, cache beta dtype, and return matching `dBeta`, but the public op validator, FROST engine admission check, and FLA adapter still rejected or rounded the official FP32 input. This PR closes that contract seam while reusing #819's implementation.

## Related issues

Related to #819.

## API and compatibility impact

This is a backward-compatible extension of the Python KDA contract. With fused beta sigmoid enabled, `beta` and `dBeta` may both be FP32 or both use the I/O dtype. Existing I/O-dtype callers and post-sigmoid FP32/I/O-dtype callers are unchanged. The adapter continues to fail closed for unsupported dtypes. This does not expand the existing architecture or software-version support of the FROST KDA path.

The feature itself is performance-neutral. In a current official-count Kimi feature-substack A/B, the optimized cuDNN arm with this contract and merged #819 measured `3839.153 -> 2829.813 ms` (`1.3567x`) at sequence 8192. That is the composed KDA + causal-conv + dense/shared SiTU result, not marginal speedup attributable to this PR.

## Testing

Latest `develop` plus this two-commit patch:

- `pre-commit run --files python/cudnn/fla/kda.py python/cudnn/linear_attention/frost/kda_engine.py python/cudnn/linear_attention/ops/kda.py test/python/linear_attention/test_fla_compat.py test/python/linear_attention/test_la.py`: passed.
- ComputeLab NVIDIA B200: `test_kda_parity_fused` plus every FROST `beta_sigmoid` case: `6 passed, 1174 deselected`. This covers FP32/BF16 raw-beta forward, matching-dtype `dBeta`, the official fused Kimi adapter route, and the FP32-calibrated output/all-gradient check.
- Official-count Kimi feature-substack at sequence 8192: seven alternating rounds, full output/input/all-parameter-gradient checks, native route proof, and the original 5% correctness caps all passed. The composed timing is reported above.

## Summary by CodeRabbit

* **New Features**
  * Expanded KDA support for mixed tensor dtypes, including FP16, BF16, and FP32 decay values, beta inputs, and initial states.
  * Added support for FP32 beta logits and gradients in fused sigmoid mode.
  * Added support for int64 sequence-length metadata.

* **Bug Fixes**
  * Improved dtype validation for forward and backward operations, including state and gradient matching.
  * Improved caching behavior across different tensor dtype combinations.

* **Tests**
  * Expanded coverage for dtype combinations, checkpointing, gradients, parity, validation, and repeatability.

### CODE REVIEW

**Comment by YangXu1990uiuc** [comment]
@cudnn-ci-bot run python_tests,frost

File: test/python/linear_attention/test_la.py [hunk 1]
```diff
@@ -1050,6 +1050,47 @@ def test_beta_sigmoid_in_kernel(backend):
     assert rms_ratio(fs_raw, fs_eff) < 2e-2
 
 
+@pytest.mark.parametrize("checkpoint", [0, 16], ids=["recompute", "checkpoint"])
```
[the remark this answers is not in the document]
Comment (YangXu1990uiuc) [reply test/python/linear_attention/test_la.py]: This thread is outdated after the rebase onto the merged linear-attention base. The reviewed checkpoint-parametrized test no longer exists on the current head; the remaining beta-sigmoid tests follow the file's existing module-level L0 convention and do not add the reviewed compiled-cache/checkpoint sweep. Resolving as no longer applicable.

**Comment by jhjpark** [comment]
Closing. Absorbed in #890.
